jclklib/client/jclk_init.cpp

```diff
@@ -171,9 +171,25 @@ bool JClkLibClientApi::jcl_disconnect()
 
 bool check_proxy_liveness(ClientState &appClientState)
 {
-    unsigned int timeout_sec = (unsigned int) DEFAULT_LIVENESS_TIME_OUT;
+    std::chrono::milliseconds timeout = std::chrono::milliseconds((unsigned int) DEFAULT_LIVENESS_TIME_OUT);
+
     Message0 connectMsg(new ClientConnectMessage());
     ClientConnectMessage *cmsg = dynamic_cast<decltype(cmsg)>(connectMsg.get());
+    struct timespec timeSpec;
+
+    if (clock_gettime(CLOCK_MONOTONIC, &timeSpec) == -1) {
+        PrintDebug("[CONNECT] Failed to get current time.\n");
+        return -1;
+    }
+
+    double timeout_second = timeSpec.tv_nsec / 1e9;
+    timeout_second += timeSpec.tv_sec;
+    timeout_second -= appClientState.get_proxy_liveness();
+
+    std::chrono::duration<double> test(timeout_second);
+
+    if (test < timeout)
+        return true;
```

Comment:
create a function with last_notification_time, current_time, and, timeout argument, and perform calculation inside.